microcosm_pubsub/producer.py

```diff
@@ -201,16 +192,16 @@ def generate_message_batches(self, messages, deferred_batch_size):
         for i in range(0, len(messages), deferred_batch_size):
             yield messages[i:i + deferred_batch_size]
 
-    def construct_batch_pubsub_message(self, message_batch: List[PubsubMessage]):
+    def construct_batch_pubsub_message(self, message_batch: List[PublishedMessage]):
         return [
             dict(
-                media_type=pubsub_message.media_type,
-                message=pubsub_message.message,
-                message_attributes=pubsub_message.message_attributes,
-                opaque_data=pubsub_message.opaque_data,
-                topic_arn=pubsub_message.topic_arn,
+                media_type=message.media_type,
+                message=message.message,
+                message_attributes=self.producer.choose_message_attributes(message.media_type),
+                opaque_data=message.opaque_data,
+                topic_arn=self.producer.choose_topic_arn(message.media_type),
```

Comment:
This implicitly means that `self.producer` has to be `sns_producer`; but that assumption was already there, since we were assuming that the message we get from `self.producer.create_message` had a `topic_arns` attribute.